File: molina-healthcare/src/main/java/base/Locator.java
package base;

import lombok.Value;

@Value
public class Locator {
    String type;   // "css" | "xpath" | "id" | "name" | "data-testid" | "aria-label"
    String value;

    /** Reconstructs the selectorKey string used in pom_registry.json for fallback lookup. */
    public String getSelectorKey() {
        switch (type) {
            case "id":          return "#" + value;
            case "name":        return "[name='" + value + "']";
            case "xpath":       return "xpath=" + value;
            case "data-testid": return "data-testid=" + value;
            case "aria-label":  return "aria-label=" + value;
            default:            return value;  // css selector stored as-is
        }
    }
}

File: cms-app-crawler-ts/src/base/Locator.ts
export type LocatorType = 'css' | 'xpath' | 'id' | 'name' | 'data-testid' | 'aria-label';

export class Locator {
  readonly type: LocatorType;
  readonly value: string;

  constructor(type: LocatorType, value: string) {
    this.type = type;
    this.value = value;
  }

  getSelectorKey(): string {
    switch (this.type) {
      case 'id':          return '#' + this.value;
      case 'name':        return `[name='${this.value}']`;
      case 'xpath':       return 'xpath=' + this.value;
      case 'data-testid': return 'data-testid=' + this.value;
      case 'aria-label':  return 'aria-label=' + this.value;
      default:            return this.value;
    }
  }

  toPlaywrightSelector(): string {
    switch (this.type) {
      case 'id':          return `#${this.value}`;
      case 'name':        return `[name='${this.value}']`;
      case 'xpath':       return `xpath=${this.value}`;
      case 'data-testid': return `[data-testid='${this.value}']`;
      case 'aria-label':  return `[aria-label='${this.value}']`;
      default:            return this.value;
    }
  }
}
